File: ejemplo2python/Vehiculo.py
class Vehiculo:
    def __init__(self, tipo, color, ancho, alto, largo):
        # Atributos de la clase
        self.tipo = tipo
        self.color = color
        self.ancho = ancho
        self.alto = alto
        self.largo = largo

    def arrancar(self):
        print(f"El vehículo {self.tipo} está arrancando.")

    def acelerar(self):
        print(f"El vehículo {self.tipo} está acelerando.")

    def frenar(self):
        print(f"El vehículo {self.tipo} está frenando.")

    # Métodos getter
    def get_color(self):
        return self.color

    def get_ancho(self):
        return self.ancho

    def get_alto(self):
        return self.alto

    def get_largo(self):
        return self.largo

    # Métodos setter
    def set_color(self, nuevo_color):
        self.color = nuevo_color

    def set_ancho(self, nuevo_ancho):
        self.ancho = nuevo_ancho

    def set_alto(self, nuevo_alto):
        self.alto = nuevo_alto

    def set_largo(self, nuevo_largo):
        self.largo = nuevo_largo

File: ejemplo1cpp/Vehiculo.cpp
#include"Vehiculo.h"


Vehiculo::Vehiculo(string tipo, string color, int ancho, int alto, int largo){ //método constructor
            this->tipo = tipo;
            this->color = color;
            this->ancho = ancho;
            this->alto = alto;
            this->largo = largo;
}

void Vehiculo::arrancar(){
    cout<<"Vehiculo  " <<tipo<<" arrancando "<<endl;
}

void Vehiculo::acelerar(){
    cout<<"Vehiculo  " <<tipo<<" acelerando "<<endl;
}


void Vehiculo::frenar(){
    cout<<"Vehiculo  " <<tipo<<" frenando "<<endl;    
}

int Vehiculo::getAncho(){
    return ancho; 
}

int Vehiculo::getAlto(){
    return alto;
}
int Vehiculo::getLargo(){
    return largo;    
}

void Vehiculo::getColor(){
    cout<<color<<endl;
}

void Vehiculo::setColor( string _newcolor){
    color = _newcolor;
}

void Vehiculo::setAlto(int _newalto){
    alto = _newalto; 
}
void Vehiculo::setAncho(int _newancho){
    ancho = _newancho;
}

void Vehiculo::setLargo(int _newlargo){
    largo = _newlargo;
}

Vehiculo::~Vehiculo(){
    cout << "Destruyendo el vehiculo de tipo: " << tipo << endl;
}
